### ISSUE TITLE
OneToOne inverse yields doctrine.associationType false positive

### ISSUE THREAD

**Issue by bobvandevijver** [phpstan/phpstan-doctrine#715]
Consider the following construct

```php
#[ORM\Entity]
class A {
  #[ORM\OneToOne(inversedBy: 'a', cascade: ['persist'], fetch: 'EXTRA_LAZY')]
  private ?B $b = null;
}

#[ORM\Entity]
class B {
  #[ORM\OneToOne(mappedBy: 'b')]
  private A $a; // Default in constructor
}
```

This results in:

```
 ------ ----------------------------------------------------------------------------------------------------------------
  Line   src/Entity/B.php
 ------ ----------------------------------------------------------------------------------------------------------------
  29     Property App\Entity\B::$a type mapping mismatch: database can contain
         App\Entity\A|null but property expects App\Entity\A.
         🪪  doctrine.associationType
 ------ ----------------------------------------------------------------------------------------------------------------
```

However, this is a false positive, as the `$a` never even makes it to the database with this construction. The `A` holds an optional reference to the `B` object.

According to doctrine, when trying to add a join column definition to disallow a `null` value.

```
App\Entity\B#a is a OneToOne inverse side, which does not allow join columns.
```

**Comment by jhae-de** [comment]
Specifying `nullable` on columns that end up being used in a primary key is depreacted since `doctrine/orm` 3.6.0. That seems to be heading in the same direction.

```php
#[ORM\Entity]
class A {
}

#[ORM\Entity]
class B {
  #[ORM\ManyToOne]
  #[ORM\JoinColumn]
  public A $a;
}
```

This results in:

```
 ------ -----------------------------------------------------------------------------------------
  Line   src/Entity/B.php                                                                                                                                                                             
 ------ -----------------------------------------------------------------------------------------
  0      Property App\Entity\B::$a type mapping mismatch: database can contain
         App\Entity\B|null but property expects App\Entity\B.  
         🪪  doctrine.associationType                                                                                                                                                                              
 ------ -----------------------------------------------------------------------------------------
```

Applies to `JoinColumn` and `InverseJoinColumn`.
